### PR TITLE
Scientific and Programmer tests

### LINKED ISSUE #608: Automated UI Tests for the Scientific Calculator Mode
**Problem Statement**
Currently, the basic smoke tests for Scientific calculator mode are all manual.  There aren't any automated UI tests for calculator modes other than the standard calculator mode.  Automation testing is faster than manual testing. Repeated test that can be automated, should be automated.

**Evidence or User Insights**
It would be helpful to developers and testers to have some streamlined, reliable UI tests for the Scientific calculator mode.  

**Proposal**
- These tests would allow developers to know that their feature/code change, when introduced to the application, does not break other aspects of the application.  
- These tests allow testers to focus on more creative, testing scenarios during manual testing rather than spending a lot of extra time executing tests that an automaton could complete.

**Goals**
- Replace basic, repeatable manual tests, with streamlined, reliable automated UI tests for the Scientific calculator mode
- Use these tests as a gate when checking in new features

**Non-Goals**
- The UI automated tests are not a replacement for Unit tests
- The UI automated tests will not cover end-to-end testing (It will cover smoke/BVT type tests)

**Low-Fidelity Concept**
Read through "ManualTests.md" (https://github.com/microsoft/calculator/blob/master/docs/ManualTests.md).  Would you prefer to execute these tests manually, or if possible, execute most of these through automation?   

**Requested Assignment**
If possible, I would like to implement this.

### PR DESCRIPTION
**Pull request by Daniel-Parker**
### Description of the changes:
- Added basic functional tests for Scientific mode to be run with every test pass

### How changes were validated:
- I ran all tests consecutively in many different orders to ensure they always pass

### CODE REVIEW

File: src/CalculatorUITests/ScientificModeFunctionalTests.cs [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+        {
+            page.ScientificOperators.AngleOperatorsButton.State = DegRadGradState.Degrees;
+
+            page.ScientificOperators.NumberPad.Input(180);
+            page.ScientificOperators.TrigButton.Click();
+            page.ScientificOperators.CosButton.Click();
+            page.ScientificOperators.EqualButton.Click();
+            Assert.AreEqual("-1", page.GetCalculatorResultText());
+
+            //page.ScientificOperators.ResetTrigDropdownToggles();
```
Comment (sanderl) [inline src/CalculatorUITests/ScientificModeFunctionalTests.cs]: Remove this if it is no longer needed. Same for the below tests.

File: src/CalculatorUITestFramework/DegRadGradButton.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                        break;
+                    case DegRadGradState.Radians:
+                        desiredId = "radButton";
+                        break;
+                    default:
+                        throw new NotImplementedException();
+                }
+                while (this.Element.GetAttribute("AutomationId") != desiredId)
+                {
+                    var foo = this.Element.GetAttribute("AutomationId");
```
Comment (EriWong) [inline src/CalculatorUITestFramework/DegRadGradButton.cs]: >foo [](start = 24, length = 3)

remove this line :)

**Comment by v-credwa** [comment]
It looks like [TestCleanup] in ScientificModeFunctionalTest.cs needs something like a return toggles to default state type of method.  Some of your tests are fail due to previous test toggling a button like the Fixed To Exponential Button.

**Comment by v-credwa** [comment]
The method "public void NavigateToCalculatorMode(CalculatorMode mode)" is recreated and used in place of specific menu navigation, in the StandardCalculatorPage.cs and in the ScientificCalculatorPage.cs without making use full use of the NavigationMenu.cs where the method was created. 

StandardCalculatorPage.cs
E.g. of using specific navigation for Standard menu navigation:
`        public void NavigateToStandardCalculator()
        {
            // Ensure that calculator is in standard mode
            this.NavigationMenu.ChangeCalculatorMode(CalculatorMode.StandardCalculator);
            this.CalculatorResults.CalculatorResultsDisplayIsPresent();
        }
`
ScientificCalculatorPage.cs
E.g. of using specific navigation for Scientific menu navigation:
`        public void NavigateToScientificCalculator()
        {
            // Ensure that calculator is in scientific mode
            this.NavigationMenu.ChangeCalculatorMode(CalculatorMode.ScientificCalculator);
            this.CalculatorResults.CalculatorResultsDisplayIsPresent();
        }
`

File: src/CalculatorUITestFramework/ProgrammerOperatorsPanel.cs [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        public WindowsElement EButton => this.session.TryFindElementByAccessibilityId("eButton");
+        public WindowsElement FButton => this.session.TryFindElementByAccessibilityId("fButton");
+        public WindowsElement LeftShiftButton => this.session.TryFindElementByAccessibilityId("lshButton");
+        public WindowsElement RightShiftButton => this.session.TryFindElementByAccessibilityId("rshButton");
+        public WindowsElement LeftShiftLogicalButton => this.session.TryFindElementByAccessibilityId("lshLogicalButton");
+        public WindowsElement RightShiftLogicalButton => this.session.TryFindElementByAccessibilityId("rshLogicalButton");
+        public WindowsElement RoLButton => this.session.TryFindElementByAccessibilityId("rolButton");
+        public WindowsElement RoRButton => this.session.TryFindElementByAccessibilityId("rorButton");
+        public WindowsElement RoRCarryButton => this.session.TryFindElementByAccessibilityId("rorCarryButton");
+        public WindowsElement Bit0 => this.session.TryFindElementByAccessibilityId("Bit0");
```
Comment (sanderl) [inline src/CalculatorUITestFramework/ProgrammerOperatorsPanel.cs:57]: Are you using all of these Bit buttons? If not, let's just include the ones that are used in the tests.

File: src/CalculatorUITestFramework/StandardAoTCalculatorPage.cs [hunk 4]
```diff
@@ -160,5 +160,43 @@ private void GrowWindowToShowInvertButton(int height)
...
             }
         }
+        /// <summary>
+        /// If the Invert button is not displayed, resize the window
+        /// Two attempts are made, the the button is not found a "not found" exception is thrown
+        /// </summary>
+        public void ResizeAoTWindowToDiplayInvertButton()
+        {
+            Point newWindowPostion = new Point(8, 8);
+            WinAppDriver.Instance.CalculatorSession.Manage().Window.Position = newWindowPostion;
```
Comment (sanderl) [inline src/CalculatorUITestFramework/StandardAoTCalculatorPage.cs:170]: you don't need multiple window position and source objects. Just reuse the first one.

File: src/CalculatorUITests/ProgrammerModeFunctionalTests.cs [hunk 5]
```diff
@@ -0,0 +1,892 @@
...
+        /// </summary>
+        [TestInitialize]
+        public void TestInit()
+        {
+            if ("0" != page.CalculatorResults.GetCalculatorResultText())
+            {
+                page.ClearAll();
+            }
+            CalculatorApp.EnsureCalculatorHasFocus();
+            page.ProgrammerOperators.SetArithmeticShift();
```
Comment (sanderl) [inline src/CalculatorUITests/ProgrammerModeFunctionalTests.cs:56]: Consider doing these in cleanup as well.

**Pull request merged**